ambry-store/src/main/java/com.github.ambry.store/BlobStore.java

```diff
@@ -150,7 +150,7 @@ private BlobStore(ReplicaId replicaId, String storeId, StoreConfig config, Sched
     this.factory = factory;
     this.recovery = recovery;
     this.hardDelete = hardDelete;
-    this.writeStatusDelegate = writeStatusDelegate;
+    this.writeStatusDelegate = config.storeWriteStatusDelegateEnable ? writeStatusDelegate : null;
```

Comment:
should we do this here or stop its creation in `StorageManager`?

Reply:
Advantage of storage manager is that it's at a broader level, disadvantage is that it would be harder to unit test.... idk